### PR TITLE
Xtend: fixed strange error message for `void` as a field type

### LINKED ISSUE #2877: Xtend: Strange error message for `void` as a field type
When you use `void` for an Xtend field the error message is

```
Primitive void cannot be a dependency.
```

Coming from `XtendValidator`

```java
	@Check
	public void checkNoVoidInDependencyDeclaration(XtendField dep) {
		JvmTypeReference declaredFieldType = dep.getType();
		if (isPrimitiveVoid(declaredFieldType)) {
			error("Primitive void cannot be a dependency.", dep.getType(), null, INVALID_USE_OF_TYPE);
		}
	}
```

Why not something similar to parameters' corresponding error message `void is an invalid type for the field... `

@szarnekow

### PR DESCRIPTION
**Pull request by LorenzoBettini**
Closes #2877

### CODE REVIEW

**Comment by LorenzoBettini** [comment]
@szarnekow  I rebased the PR, which is now ready to review

File: org.eclipse.xtend.core/src/org/eclipse/xtend/core/validation/XtendValidator.java [hunk 1]
```diff
@@ -500,10 +500,11 @@ protected EObject getContainingAnnotationTarget(XAnnotation annotation) {
 	}
 
 	@Check
-	public void checkNoVoidInDependencyDeclaration(XtendField dep) {
-		JvmTypeReference declaredFieldType = dep.getType();
+	public void checkXtendFieldNotPrimitiveVoid(XtendField field) {
+		JvmTypeReference declaredFieldType = field.getType();
 		if (isPrimitiveVoid(declaredFieldType)) {
-			error("Primitive void cannot be a dependency.", dep.getType(), null, INVALID_USE_OF_TYPE);
+			error("void is an invalid type for the field " + field.getName(),
```
Comment (szarnekow) [inline org.eclipse.xtend.core/src/org/eclipse/xtend/core/validation/XtendValidator.java:506]: I see that the message was take from JDT. In Xbase, we use 
```
@Check
	public void checkTypeReferenceIsNotVoid(XExpression expression) {
		for (EObject eObject : expression.eContents()) {
			if (eObject instanceof JvmTypeReference) {
				JvmTypeReference typeRef = (JvmTypeReference) eObject;
				if (isPrimitiveVoid(typeRef)) {
					if (typeRef.eClass() == TypesPackage.Literals.JVM_PARAMETERIZED_TYPE_REFERENCE) {
						if (!((JvmParameterizedTypeReference) typeRef).getArguments().isEmpty()) {
							continue;
						}
					}
					error("Primitive void cannot be used here.", typeRef, null, INVALID_USE_OF_TYPE);
				}
			}
		}
	}
```

and in common.types

```
	protected void checkNotPrimitive(JvmTypeReference jvmTypeReference) {
		if (primitives.isPrimitive(jvmTypeReference)) {
			error("The primitive '"+jvmTypeReference.getQualifiedName('.')+"' cannot be a type argument", jvmTypeReference, null, IssueCodes.INVALID_USE_OF_TYPE);
		}
	}
```

All not super aligned with each other. Maybe keep it somewhat consistent and use `Primitive void is an invalid type for the field ...`

**Review by szarnekow — APPROVED** [review]

File: org.eclipse.xtend.core/src/org/eclipse/xtend/core/validation/XtendValidator.java (hunk 1 above)
Comment (LorenzoBettini) [reply org.eclipse.xtend.core/src/org/eclipse/xtend/core/validation/XtendValidator.java:506]: @szarnekow OK, I'll change it to `Primitive void is an invalid type for the field` and then merge it, OK?

**Comment by LorenzoBettini** [comment]
@szarnekow , I checked other points in the XtendValidator:

```java
	@Check
	public void checkXtendParameterNotPrimitiveVoid(XtendParameter param) {
		if (isPrimitiveVoid(param.getParameterType())) {
			XtendFunction function = (XtendFunction) (param.eContainer() instanceof XtendFunction ? param.eContainer()
					: null);
			if (function != null)
				error("void is an invalid type for the parameter " + param.getName() + " of the method "
						+ function.getName(), param.getParameterType(), null, INVALID_USE_OF_TYPE);
			else
				error("void is an invalid type for the parameter " + param.getName(), param.getParameterType(), null,
						INVALID_USE_OF_TYPE);
		}
	}
```

we always use that form "void is an invalid type for..." also in other parts:

image

So, shall I leave it like that?

**Pull request merged**